src/helpers/filterExpenses.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+function filterExpenses(expenses, categories, userId, from, to) {
+  let filteredExpenses = [...expenses];
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(
+      expense => categories.includes(expense.category)
+    );
+  }
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(
+      expense => expense.userId === +userId
+    );
+  }
+
+  if (from && to) {
+    const toDate = new Date(to);
```

Comment:
You don't need to filter expenses, just query from the DB what you actually need